File: Multi-Threading-Python/arrayGenerator.py
import random

def generate_random_array(array_size):
    """
    Generate an array of random integers.

    Parameters:
        array_size (int): The size of the array to generate.

    Returns:
        list: A list of random integers between 0 and 10000.

    Raises:
        ValueError: If array_size is not a non-negative integer.
    """

    # Input validation
    if not isinstance(array_size, int):
        raise ValueError("Array size must be an integer")

    if array_size < 0:
        raise ValueError("Array size must be a non-negative integer")

    # Generate and return a list of random integers
    return [random.randint(0, 10000) for _ in range(array_size)]

File: javaProject/ArrayGenerator.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class ArrayGenerator {
	
	/**
	 * Generate an array of random integers
	 * 
	 * @param size of the array
	 * @return List<Integer> of random integers between 0 and 10000 
	 */
	public List<Integer> getRandomArray(int size) {
		// Input validation
		if (size < 0) {
			throw new IllegalArgumentException("Array size must be a non-negative number");
		}
		
		List<Integer> elements = new ArrayList<>(size);
		Random r = new Random();
		
		// Generate random elements
		for(int i = 0; i < size; i++) {
			elements.add(r.nextInt(10001));
		}
		
		return elements;
	}
}
